### ISSUE TITLE
Bug in fit function

### ISSUE THREAD

**Issue by crstnbr** [JuliaMath/Polynomials.jl#224]
Polynomials v1.0.6 and Julia 1.4.1.

```julia
julia> using Polynomials

julia> fit(Polynomial, rand(2,2), rand(2,2))
[1]    17190 bus error  Julia
```

**Comment by jverzani** [comment]
Yes, that looks scary.  The code path  is  expecting `promote(collect(x),collect(y))` to be a pair of   abstract vectors to   be handled through  dispatch, and here  you have matrices. Before  trying  to   work around this, what did  you expect to have as  the output?

**Comment by crstnbr** [comment]
> what did you expect to have as the output?

I would have expected an error since the input arguments - the x and y points - aren't vectors.

**Comment by jverzani** [comment]
closed by #226
